care/facility/models/bed.py

```diff
@@ -51,6 +51,10 @@ def save(self, *args, **kwargs) -> None:
         self.validate()
         return super().save(*args, **kwargs)
 
+    def delete(self, *args, **kwargs) -> None:
+        super().delete(*args, **kwargs)
+        AssetBed.objects.filter(bed=self).update(deleted=True)
```

Comment:
AssetBed objects should be deleted before because they have models.PROTECT flag set on Bed FK